File: task4.py
def snake_to_camel(snake_dict):
    camel_dict = {}
    for key, value in snake_dict.items():
        # Split the key by underscores, capitalize letters after the first word, and join them back
        camel_key = ''.join([word.capitalize() if i > 0 else word for i, word in enumerate(key.split('_'))])
        # Make the first character lowercase
        camel_dict[camel_key[0].lower() + camel_key[1:]] = value
    return camel_dict

# Example usage
snake_case_json = {
    "first_name": "John",
    "last_name": "Doe",
    "email_address": "john.doe@example.com"
}

camel_case_json = snake_to_camel(snake_case_json)
print(camel_case_json)

File: task4.js
function snakeToCamel(snakeObj) {
    let camelObj = {};
    for (let key in snakeObj) {
        if (snakeObj.hasOwnProperty(key)) {
            // Replace underscores and capitalize the next letter
            let camelKey = key.replace(/_([a-z])/g, (match, letter) => letter.toUpperCase());
            // Make the first letter lowercase to follow camelCase
            camelObj[camelKey[0].toLowerCase() + camelKey.slice(1)] = snakeObj[key];
        }
    }
    return camelObj;
}

// Example usage
const snakeCaseJson = {
    "first_name": "John",
    "last_name": "Doe",
    "email_address": "john.doe@example.com"
};

const camelCaseJson = snakeToCamel(snakeCaseJson);
console.log(camelCaseJson);
